### PR TITLE
Fix dev guides

### LINKED ISSUE #1239: Improve guides about generating module/plugin/command
### What
E.g. https://lisk.com/documentation/lisk-sdk/guides/app-development/module.html#generating-the-module-skeleton

### How
Explain which characters can not be used as alias when using Lisk Commander to generate them.

### Additional information
Hi, HQ devs! It's probably only a small thing but just wanted to let you know. When using $ lisk generate:plugin, I suppose you cannot use - in the Alias arg.

### LINKED ISSUE #1803: Issues in Hello World guides
### Description

- The custom `MessageStoreData` type should use `string` instead of `String`
- When registering the custom stores, no index was provided as an argument of the constructor
- In step 3 of How to create a module, it is not shown how an event is made, yet it is being used in the code examples. Link should be added here to step 5 where it is explained.
- `counterKey` is not mentioned in the guide / code examples in chapter 2
- `curl` is being used to interact with the HTTP API. Would be better to use the sidechain's CLI?
- When testing the `createHello` command (with valid data), the following error is returned: Cannot read properties of undefined (reading 'status'). Thats because docs are missing description of transaction validation 

```
public async verifyTransaction(context: TransactionVerifyContext): Promise<VerificationResult> {
        const result = {
            status: 1,
        };
        return result;
    }
```

### Affected branch / URL
`development`

### PR DESCRIPTION
**Pull request by Tschakki**
Closes #1803 
Closes #1239

### CODE REVIEW

File: docs/modules/ROOT/pages/build-blockchain/module/command.adoc [hunk 1]
```diff
@@ -124,6 +126,23 @@ export class HelloModule extends BaseModule {
...
 ----
 
+[NOTE]
+====
+To avoid errors when <<try-the-new-command-out, trying out>> the new command, adjust the `verifyTransaction()` hook of the module to the following:
+
+[source,typescript]
+----
+public async verifyTransaction(context: TransactionVerifyContext): Promise<VerificationResult> {
+    // verify transaction will be called multiple times in the transaction pool
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/module/command.adoc]:
```suggestion
    // Verify transaction will be called multiple times in the transaction pool
```

File: docs/modules/ROOT/pages/build-blockchain/module/endpoints-methods.adoc [hunk 2]
```diff
@@ -179,16 +179,7 @@ export class HelloMethod extends BaseMethod {
...
-    "id": "1",
-    "method": "hello_getHello",
-    "params": {
-        "address": "lskuz5p98kz3mqzxnu68qdrjxtvdvr2o7pprtj4yv"
-    }
-}'
+./bin/run endpoint:invoke hello_getHello '{"address": "lskuz5p98kz3mqzxnu68qdrjxtvdvr2o7pprtj4yv"}'
 ----
 +
 When the request is successful, the node will reply with the following:
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/module/endpoints-methods.adoc:185]: The response on lines **187-196** should also be aligned with the CLI request format. I get the following response when I query with CLI.

image

File: docs/modules/ROOT/pages/build-blockchain/module/endpoints-methods.adoc [hunk 3]
```diff
@@ -208,15 +199,7 @@ When the request is successful, the node will reply with the following:
...
-    "jsonrpc": "2.0",
-    "id": "1",
-    "method": "hello_getHelloCounter",
-    "params": {
-    }
-}'
+./bin/run endpoint:invoke hello_getHelloCounter
 ----
 +
 When the request is successful, the node will return the current counter value:
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/module/endpoints-methods.adoc:199]: Same comment as above about the CLI response.

File: docs/modules/ROOT/pages/build-blockchain/module/index.adoc [hunk 4]
```diff
@@ -91,6 +93,8 @@ As an example, we use the module name `hello`:
 lisk generate:module hello
 ----
 
+NOTE: The module name should contain only characters from the set `[a-z0-9!@$&_.]`.
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/module/index.adoc]: I would make it bold to make it more pronounced.
```suggestion
NOTE: The module name should contain only characters from the set `*[a-z0-9!@$&_.]*`.
```

File: docs/modules/ROOT/pages/build-blockchain/module/stores.adoc [hunk 5]
```diff
@@ -190,8 +192,8 @@ export class HelloModule extends BaseModule {
     public constructor() {
         super();
         // registration of stores and events
-        this.stores.register(CounterStore, new CounterStore(this.name));
-        this.stores.register(MessageStore, new MessageStore(this.name));
+        this.stores.register(CounterStore, new CounterStore(this.name, 0));
+		this.stores.register(MessageStore, new MessageStore(this.name, 1));
```
Comment (TalhaMaliktz) [inline docs/modules/ROOT/pages/build-blockchain/module/stores.adoc]:
```suggestion
        this.stores.register(CounterStore, new CounterStore(this.name, 0));
	this.stores.register(MessageStore, new MessageStore(this.name, 1));
```

**Review by TalhaMaliktz — CHANGES_REQUESTED** [review]
Looks good already!

File: docs/modules/ROOT/pages/build-blockchain/module/endpoints-methods.adoc (hunk 3 above)
Comment (Tschakki) [reply docs/modules/ROOT/pages/build-blockchain/module/endpoints-methods.adoc:199]: updated 👍

**Review by ishantiw — APPROVED** [review]

**Review by TalhaMaliktz — APPROVED** [review]
Thank you! LGTM 👍🏼

**Pull request merged**